### PR TITLE
✨ Reduce some clutter in v1a2 Status.Network

### PR DESCRIPTION
**Pull request by bryanv**
Try to not show interfaces and IPs that aren't informative. On TKG nodes especially there will a lot of pseudo interface and routes that just make the output hard to read.

This is just the first pass and we'll likely need more refinement later.

```release-note
In the v1a2 Status.Network, certain devices and routes are not shown to reduce clutter. Specifically, pseudo devices and link and interface local routes are skipped.
```

### CODE REVIEW

File: pkg/vmprovider/providers/vsphere2/vmlifecycle/update_status.go [hunk 1]
```diff
@@ -267,9 +266,31 @@ func convertNetIPRouteConfigInfo(routeConfig *types.NetIpRouteConfigInfo) []vmop
...
+
+		if ip.To4() != nil {
+			return prefix == 32
+		}
+
+		ip6 := ip.To16()
+		return ip6 == nil || ip6.IsLinkLocalUnicast() || ip6.IsMulticast()
+	}
+
+	out := make([]vmopv1.VirtualMachineNetworkIPRouteStatus, 0, 1)
```
Comment (bryanv) [inline pkg/vmprovider/providers/vsphere2/vmlifecycle/update_status.go:287]: Silence linter wanting for this to be preallocated. Common case will be 1'ish.

File: pkg/vmprovider/providers/vsphere2/vmlifecycle/update_status.go [hunk 2]
```diff
@@ -267,9 +266,31 @@ func convertNetIPRouteConfigInfo(routeConfig *types.NetIpRouteConfigInfo) []vmop
 		return nil
 	}
 
-	// TODO: Prob only want to show default routes. Will be very verbose on TKG nodes.
-	out := make([]vmopv1.VirtualMachineNetworkIPRouteStatus, 0, len(routeConfig.IpRoute))
+	// Try to skip uninteresting routes so the Status isn't as cluttered.
```
Comment (akutz) [inline pkg/vmprovider/providers/vsphere2/vmlifecycle/update_status.go:269]: I am fine skipping loopback addresses, but I do think static routes should be shown. There are often non-default routes.

**Comment by akutz** [comment]
Can we please update the PR description and release note to indicate what data is being omitted/optimized away? I should not have to look at the code to know what is being dropped / retained.

**Review by aruneshpa — APPROVED** [review]
LGTM.

File: pkg/vmprovider/providers/vsphere2/vmlifecycle/update_status.go (hunk 2 above)
Comment (akutz) [reply pkg/vmprovider/providers/vsphere2/vmlifecycle/update_status.go:269]: Let me be more clear, I'm also fine skipping link-local routes.

**Review by akutz — APPROVED** [review]
Approved after @bryanv and I spoke offline. He will add a comment to the code that filters the routes, explaining what is being filtered. Thanks BV!

**Pull request merged**